instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/runner/DatadogTestsRunner.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.nightly.runner
+
+import android.os.Bundle
+import android.util.Log
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.runner.AndroidJUnitRunner
+import com.datadog.android.Datadog
+import com.datadog.android.core.configuration.Configuration
+import com.datadog.android.core.configuration.Credentials
+import com.datadog.android.nightly.BuildConfig
+import com.datadog.android.privacy.TrackingConsent
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumMonitor
+import com.datadog.android.tracing.AndroidTracer
+import com.datadog.tools.unit.invokeMethod
+import io.opentracing.util.GlobalTracer
+import java.util.concurrent.TimeUnit
+
+class DatadogTestsRunner() : AndroidJUnitRunner() {
+
+    // region Runner
+
+    override fun start() {
+        initializeDatadog()
+        super.start()
+    }
+
+    override fun finish(resultCode: Int, results: Bundle?) {
+        Thread.sleep(DEFAULT_TIMEOUT_IN_MS)
```

Comment:
if I understand the doc of `AndroidJUnitRunner` correctly, `start` and `finish` correspond to the lifecycle of the instrumentation itself and not to the individual tests lifecycle, so this method will be called only once __all__ tests are over and we won't have a delay of 1m after each test?

Reply:
yes, that's  the goal of it.